crates/bevy_ecs/src/archetype.rs

```diff
@@ -342,46 +481,60 @@ impl Archetype {
             .map(|info| info.archetype_component_id)
     }
 
+    /// Clears all entities from the archetype.
     pub(crate) fn clear_entities(&mut self) {
         self.entities.clear();
     }
 }
 
-/// A generational id that changes every time the set of archetypes changes
-#[derive(Debug, Copy, Clone, Eq, PartialEq)]
+/// An opaque generational id that changes every time the set of [`Archetypes`] changes.
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Ord, PartialOrd)]
 pub struct ArchetypeGeneration(usize);
 
 impl ArchetypeGeneration {
     #[inline]
-    pub const fn initial() -> Self {
+    pub(crate) const fn initial() -> Self {
         ArchetypeGeneration(0)
     }
 
     #[inline]
-    pub fn value(self) -> usize {
+    pub(crate) fn value(self) -> usize {
         self.0
     }
 }
 
 #[derive(Hash, PartialEq, Eq)]
-pub struct ArchetypeIdentity {
+struct ArchetypeIdentity {
     table_components: Box<[ComponentId]>,
     sparse_set_components: Box<[ComponentId]>,
 }
 
+/// An opaque unique joint ID for a [`Component`] in an [`Archetype`] within a [`World`].
```

Comment:
❓ What is a joint id?

Reply:
It can sort of be thought of as a composite key in SQL terms, but not sure how to express this succinctly.

Reply:
Added a paragraph about the many-to-many relationship between components and archetypes.

Reply:
Yeah, that helped me wrap my head around it! Thank you